backend/sozisel/lib/sozisel/model/event_results.ex

```diff
@@ -29,4 +30,31 @@ defmodule Sozisel.Model.EventResults do
   def change_event_result(%EventResult{} = event_result, attrs \\ %{}) do
     EventResult.create_changeset(event_result, attrs)
   end
+
+  def get_all_event_results(%LaunchedEvent{id: launched_event_id}) do
+    from(res in EventResult, where: res.launched_event_id == ^launched_event_id)
+    |> Repo.all()
+  end
+
+  def quiz_result_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+    # IO.inspect(quiz_results)
+
+    Enum.each(quiz_results, fn event_data -> 
+      %{result_data: result_data} = Utils.from_deep_struct(event_data) 
```

Comment:
You know that doing pattern matching on a single key is just enough? There is no need to do `Map.take`. Pattern matching will ignore all other fields in the map and just take the result data.

Reply:
And you complicate this code too much. I would do something like this
```elixir
Enum.map(quiz_results, & &1.result_data.quiz_time)
```

Reply:
Okey, I remember that :D